### PR TITLE
Govcloud irsa support

### PR DESCRIPTION
**Pull request by infa-mhadiman**
This adds env AWS_REGION based on annotation, vault.hashicorp.com/auth-config-region. This is needed by vault to identify cluster OIDC provider on EKS and enables OIDC auth with aws method on gov-clouds.

Fixes - #244

### CODE REVIEW

**Comment by tvoran** [comment]
It looks like this could work, but how about just adding `AWS_DEFAULT_REGION` to the list of environment variables copied from the app containers here: https://github.com/hashicorp/vault-k8s/blob/4fcb49e8fe4249e112d9b075f89718ffe4f36e1b/agent-inject/agent/agent.go#L654-L654

That would seem to address the issue in #244.

**Comment by infa-mhadiman** [comment]
hmm, I don't see this variable being injected on EKS 1.16, may be it is doing in higher versions. I'll also add the logic to copy the env AWS_DEFAULT_REGION.

**Comment by tvoran** [comment]
Was your pod output in #244 not from EKS 1.16? It's there in the mybot container:

```
Environment:
      AWS_DEFAULT_REGION:           us-gov-east-1
      AWS_ROLE_ARN:                 <REDACTED>
      AWS_WEB_IDENTITY_TOKEN_FILE:  /var/run/secrets/eks.amazonaws.com/serviceaccount/token
```

**Comment by infa-mhadiman** [comment]
Yes, we are using 1.18 on gov clouds.

**Comment by infa-mhadiman** [comment]
@tvoran, PR is updated with the changes, this should work on all the versions.

**Review by tvoran — COMMENTED** [review]
This is looking good to me, just a small suggestion for one of the test functions.

File: agent-inject/agent/container_env_test.go [hunk 1]
```diff
@@ -60,11 +60,61 @@ func TestContainerEnvsForIRSA(t *testing.T) {
...
+			Value: "default-region",
+		},
+		{
+			Name:  "AWS_REGION",
+			Value: "test-region",
+		},
+	})
+}
+
+func testPodWithRegionAnnotation() *corev1.Pod {
```
[the remark this answers is not in the document]
Comment (infa-mhadiman) [reply agent-inject/agent/container_env_test.go]: Yes,  this is fixed. No annotation is passed here. Test to check for ENV when region is passed in the authConfig.

**Review by tvoran — APPROVED** [review]
Thanks!

**Pull request merged**